ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/PeerSelection.hs

```diff
@@ -3426,6 +3428,77 @@ prop_governor_uses_ledger_peers env =
      in counterexample (intercalate "\n" $ map show $ usesLedgerPeers)
       $ all snd usesLedgerPeers
 
+
+prop_governor_promiscuous_mode :: GovernorMockEnvironment -> Property
+prop_governor_promiscuous_mode env =
+    let events = Signal.eventsFromListUpToTime (Time (10 * 60 * 60))
+               . selectPeerSelectionTraceEvents
+               . runGovernorInMockEnvironment
+               $ env
+
+        counters :: Signal (PeerSelectionSetsWithSizes PeerAddr)
+        counters =
+          selectGovState peerSelectionStateToView events
+
+        -- accumulate local roots
+        localRoots :: Signal (Set PeerAddr)
+        localRoots =
+            Signal.keyedUntil
+              (\case
+                Just (GovernorEvent (TraceLocalRootPeersChanged a _)) -> LocalRootPeers.keysSet a
+                Just (MockEnvEvent (TraceEnvSetLocalRoots a)) -> LocalRootPeers.keysSet a
+                _ -> Set.empty
+              )
+              (\_ -> Set.empty)
+              (\_ -> False)
+          . Signal.fromChangeEvents Nothing
+          . fmap Just
+          $ events
+
+        publicRoots :: Signal (Set PeerAddr)
+        publicRoots =
+            Signal.keyedUntil
+              PublicRootPeers.toSet
+              (\_ -> Set.empty)
+              (\_ -> False)
+          . selectGovState Governor.publicRootPeers
+          $ events
+
+        promiscuousMode :: Signal Bool
+        promiscuousMode =
+            Signal.fromChangeEvents True
+          $ selectGovPromiscuousMode events
+
+    in counterexample (intercalate "\n" $ show <$> Signal.eventsToList events)
+     $ signalProperty 20 show
+        (\(cs, localRootSet, publicRootSet, pm) ->
+          if not pm
+            then
+            -- we need to remove local and public roots.  They are changing
+            -- over time, and a node might keep using them, event though the
+            -- node is configured as a non-promiscuous node.
+            --
+            -- This makes this test only effective if a node starts in
+            -- non-promiscuous mode, until it is reconfigured.  This can
+            -- discover some bugs in `readPromiscuousMode` but certainly not
+            -- all.
+            --
+            -- TODO: write a more effective test.
+               Set.null (fst (viewKnownBootstrapPeers cs)
+                          Set.\\ localRootSet
+                          Set.\\ publicRootSet)
+            && Set.null (fst (viewKnownBigLedgerPeers cs)
+                          Set.\\ localRootSet
+                          Set.\\ publicRootSet)
+            && Set.null (fst (viewKnownSharedPeers cs)
+                          Set.\\ localRootSet
+                          Set.\\ publicRootSet)
+          else True)
```

Comment:
```suggestion
          pm 
              -- we need to remove local and public roots.  They are changing
              -- over time, and a node might keep using them, event though the
              -- node is configured as a non-promiscuous node.
              --
              -- This makes this test only effective if a node starts in
              -- non-promiscuous mode, until it is reconfigured.  This can
              -- discover some bugs in `readPromiscuousMode` but certainly not
              -- all.
              --
              -- TODO: write a more effective test.
          ||   (Set.null (fst (viewKnownBootstrapPeers cs)
                          Set.\\ localRootSet
                          Set.\\ publicRootSet)
            && Set.null (fst (viewKnownBigLedgerPeers cs)
                          Set.\\ localRootSet
                          Set.\\ publicRootSet)
            && Set.null (fst (viewKnownSharedPeers cs)
                          Set.\\ localRootSet
                          Set.\\ publicRootSet))
```